pkg/xpkg/lint.go

```diff
@@ -71,6 +76,42 @@ func IsConfiguration(o runtime.Object) error {
 	return nil
 }
 
+// PackageCrossplaneCompatible checks that the current Crossplane version is
+// compatible with the package constraints.
+func PackageCrossplaneCompatible(v version.Operations) parser.ObjectLinterFn {
+	return func(o runtime.Object) error {
+		p, ok := o.(pkgmeta.Pkg)
+		if !ok {
+			return errors.New(errNotMeta)
+		}
+		if p.GetCrossplaneVersion() != nil {
```

Comment:
Nit: You could return early here and below `if p.GetCrossplaneVersion() == nil `to avoid indenting the bulk of this function.